pennylane/resource/first_quantization.py

```diff
@@ -589,3 +638,353 @@ def qubit_cost(n, eta, omega, error, br=7, charge=0):
 
         return int(np.ceil(qubits))
+
+    @staticmethod
+    def _norm_noncubic(
+        n,
+        eta,
+        error,
+        br,
+        charge,
+        vectors,
+    ):
+        r"""Return the 1-norm of a first-quantized Hamiltonian in the plane-wave basis
+        for non-cubic systems.
+
+        Args:
+            n (int): number of plane waves
+            eta (int): number of electrons
+            error (float): target error in the algorithm
+            br (int): number of bits for ancilla qubit rotation
+            charge (int): total electric charge of the system
+            vectors (array[float]): lattice vectors
+
+        Returns:
+            float: 1-norm of a first-quantized Hamiltonian in the plane-wave basis
+        """
+        omega = np.abs(np.sum((np.cross(vectors[0], vectors[1]) * vectors[2])))
+
+        recip_vectors = (
+            2
+            * np.pi
+            / omega
+            * np.array([np.cross(vectors[i], vectors[j]) for i, j in [(1, 2), (2, 0), (0, 1)]])
+        )
+
+        bbt = np.matrix(recip_vectors) @ np.matrix(recip_vectors).T
+
+        orthogonal = (
+            np.linalg.norm(bbt - np.array([np.max(b**2) for b in recip_vectors]) * np.identity(3))
+            < 1e-6
+        )
+
+        l_z = eta + charge
+
+        alpha = 0.0248759298  # optimal value for lower resource estimates
+        p_th = 0.95  # optimal value for lower resource estimates
+
+        error_uv = alpha * error
+
+        # taken from Eq. (22) of PRX Quantum 2, 040332 (2021)
+        n_p = int(np.ceil(np.log2(n ** (1 / 3) + 1)))
+
+        n0 = n ** (1 / 3)
+
+        # defined in Eq. (F3) of arXiv:2302.07981 (2023)
+        bmin = np.min(np.linalg.svd(recip_vectors)[1])
+
+        n_m = int(
+            np.ceil(
+                np.log2(  # taken from Eq. (3) of arXiv:2302.07981 (2023)
+                    (8 * np.pi * eta)
+                    / (error_uv * omega * bmin**2)
+                    * (eta - 1 + 2 * l_z)
+                    * (7 * 2 ** (n_p + 1) - 9 * n_p - 11 - 3 * 2 ** (-1 * n_p))
+                )
+            )
+        )
+
+        lambda_nu = (  # expression is taken from Eq. (F6) of PRX 8, 011044 (2018)
+            4 * np.pi * (np.sqrt(3) * n ** (1 / 3) / 2 - 1)
+            + 3
+            - 3 / n ** (1 / 3)
+            + 3 * integrate.nquad(lambda x, y: 1 / (x**2 + y**2), [[1, n0], [1, n0]])[0]
+        ) / bmin**2
+
+        # computed using error term derived in Eq. (113) of PRX Quantum 2, 040332 (2021)
+        lambda_nu_1 = lambda_nu + 4 / (2**n_m * bmin**2) * (
+            7 * 2 ** (n_p + 1) - 9 * n_p - 11 - 3 * 2 ** (-1 * n_p)
+        )
+
+        # taken from Eq. (I7) of arXiv:2302.07981 (2023)
+        p_nu = lambda_nu_1 * bmin**2 / 2 ** (n_p + 6)
+
+        # amplitude amplification steps, taken from Eq. (117) of PRX Quantum 2, 040332 (2021)
+        p_nu_amp = 0
+        aa_steps = 0
+        for i in range(29, -1, -1):
```

Comment:
How this range is selected? Do we have a reference?

Reply:
We have to compute the different probabilities after `i` many steps amplitude amplification and check which is the first one (smallest `i` ) for which the amplified success probability is larger than the threshold `p_th`. 
Depending on how small the initial success probability is , we may need a larger `i`. I have never seen `i` even approach 10 in any use case, but this is just in case. There is no reference, nor is there any need for a reference for this range selection.